File: Modulo3/POO/poo.js
class User {
  constructor(name, age, email, password) {
    this.name = name;
    this.age = age;
    this.email = email;
    this.password = password;
    this.isLoggedIn = false;
  }

  // Metodos

  login(email, password) {
    if (this.email === email && this.password === password) {
      this.isLoggedIn = true;
      return "Login successful!";
    } else {
      return "Incorrect email or password.";
    }
  }

  logout() {
    this.isLoggedIn = false;
    return "Logout successful!";
  }
}

user1 = new User("Alice", 30, "alice@example.com", "password123");
user2 = new User("Bob", 25, "bob@example.com", "password456");

console.log(user1.email);
console.log(user1.login("alice@example.com", "password123"));
console.log(user2.login("bob@example.com", "password456"));
console.log(user1.login("alice@example.com", "wrongpassword"));

File: Modulo3/POO/poo.rb
class User
  attr_accessor :name, :email, :age

  def initialize(name, email, age, password)
    @name = name
    @email = email
    @age = age
    @password = password
    @is_logged_in = false
  end

  # Métodos

  def print_password
    @password
  end

  def login(email, password)
    if @email == email && @password == password
      @is_logged_in = true
      "Login successful!"
    else
      "Incorrect email or password."
    end
  end

  def logout
    @is_logged_in = false
    "Logout successful!"
  end
end

user1 = User.new("Alice", "alice@example.com", 30, "password123")
user2 = User.new("Bob", "bob@example.com", 25, "password456")

puts user1.email
puts user1.print_password
puts user1.login("alice@example.com", "password123")
puts user2.login("bob@example.com", "password456")
puts user1.login("alice@example.com", "wrongpassword")